File: 1894-merge-strings-alternately/merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int i=0;
        string merged;
        while (i < word1.length() || i < word2.length()){
            bool x = (i < word1.length()) ? true : false;
            bool y = (i < word2.length()) ? true : false;

            if (x) merged += word1[i];
            if (y) merged += word2[i];
            i+=1;
        }

        return merged;
    }
};

File: 1894-merge-strings-alternately/merge-strings-alternately.go
func mergeAlternately(word1 string, word2 string) string {
    i, j := 0, 0
    var result string
    for i < len(word1) && j < len(word2) {
        if i <= j {
            result += string(word1[i])
            i+=1
        } else {
            result += string(word2[j])
            j+=1
        }
    }

    for i < len(word1) {
        result += string(word1[i])
        i+=1
    }
    for j < len(word2) {
        result += string(word2[j])
        j+=1
    }

    return result
}